### ISSUE TITLE
no-unused-vars throws error for used variables

### ISSUE THREAD

**Issue by jankhunter** [eslint/eslint#10869]
**Tell us about your environment**

* **ESLint Version:** - v5.5.0
* **Node Version:** - v8.9.0
* **npm Version:** - 5.8.0

**What parser (default, Babel-ESLint, etc.) are you using?** - default

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
var transistionHtml;
	$.each(nodeTransitionData, function( index, node ) {
		var transtionName = node.Name;
		var transtionId = node.TransitionId;
		transistionHtml += "bala ";
	});
```

```bash
eslint filename
```

**What did you expect to happen?**
Thought this code is fine.

**What actually happened? Please include the actual, raw output from ESLint.**
Throwed error for transistionHtml.

**Comment by platinumazure** [comment]
Hi @jankhunter, thanks for the issue.

The rule expects variables to be read (e.g., on the right hand side of an assignment, or passed as a function call argument, etc.), rather than just modified.

Your code is equivalent to the following:

```js
// vat transitionHtml;
	$.each(nodeTransitionData, function( index, node ) {
		var transtionName = node.Name;
		var transtionId = node.TransitionId;
		// transistionHtml += "bala ";
	});
```

This is because transitionHtml isn't used after it is built. So no-unused-vars is correctly identifying that it's not really used (it's just modified repeatedly).

Simply use transitionHtml later in your file (read it, pass it into a jQuery object's .html() function, etc.) and the lint error should go away.
